docs/ops/movement/Gather_7.md

```diff
@@ -10,18 +10,19 @@ TensorFlow\* [Gather](https://www.tensorflow.org/api_docs/python/tf/gather) oper
 
 **Detailed description**
 
-    output[p_0, p_1, ..., p_{axis-1}, p_axis, ..., p_{axis + k}, ...] = 
-       data[p_0, p_1, ..., p_{axis-1}, indices[p_0, p_1, ..., p_{b-1}, p_b, ..., p_{axis}, j], ...]
+    output[p_0, p_1, ..., p_{axis-1}, i_b, ..., i_{M-1}, p_{axis+1}, ..., p_{N-1}] = 
+       data[p_0, p_1, ..., p_{axis-1}, indices[p_0, p_1, ..., p_{b-1}, i_b, ..., i_{M-1}], p_{axis+1}, ..., p_{N-1}]
 
-Where `data`, `indices` and `axis` are tensors from first, second and third inputs correspondingly, and `b` is 
-the number of batch dimensions.
+Where `data`, `indices` and `axis` are tensors from first, second and third inputs correspondingly, `b` is 
+the number of batch dimensions. `N` and `M` are numbers of dimensions of `data` and `indices` tensors, respectively.
 
 **Attributes**:
 * *batch_dims*
   * **Description**: *batch_dims* (also denoted as `b`) is a leading number of dimensions of `data` tensor and `indices` 
   representing the batches, and *Gather* starts to gather from the `b` dimension. It requires the first `b` 
-  dimensions in `data` and `indices` tensors to be equal.
-  * **Range of values**: `[0; min(data.rank, indices.rank))` and `batch_dims <= axis`
+  dimensions in `data` and `indices` tensors to be equal. If *batch_dims* is less than zero used normalized value 
+  `batch_dims = indices.rank + batch_dims`.
+  * **Range of values**: `[-min(data.rank, indices.rank); min(data.rank, indices.rank))` and `batch_dims <= axis`
```

Comment:
batch_dims<=axis is true for any batch_dims<0 and axis>0. Also should we require batch_dims<0 for axis<0?
 I think this condition should be apllied to normalized batch_dims value and abs(axis).

Reply:
you are right `batch_dims<=axis` is true either if both batch_dims and axis are of the same sight or if they are normalized. Will correct this

Reply:
added clarification about this, please review

Reply:
> Allowed values are from [-len(indices.shape), len(indices.shape) - 1] and axis >= batch_dims as in tensorflow

indeed there must be `data.shape` not indices. Axis value is not bound to indices. In TF for axis allowed values are 
Allowed values are from `[-len(data.shape), len(data.shape) - 1] and axis >= batch_dims`

https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/kernels/gather_op.cc#L81-L89
there params goes for 1st input in our case means data